src/components/PageSwitch.tsx

```diff
@@ -91,6 +91,10 @@ const PageSwitch = ({
     }
   }, [currentPage, setPageNumber, totalNumberOfPages])
 
+  useEffect(() => {
+    setManuallySetPage(currentPage)
+  }, [currentPage])
```

Comment:
@mvaivre this is to ensure that when the page is set through the URL, the same page is reflected when clicking to set it manually.

Reply:
Mhh, I'm not sure that this is necessary?

Reply:
What happens if you manuall a URL that has `?p=1000000000000000000000`, wait for the page to load, and then click the paginator? Without this effect, it will display `1000000000000000000000` instead of the actual number of the last page.

Reply:
Ah I see! All clear now.